src/compat/db_json.cpp

```diff
@@ -3084,6 +3139,10 @@ db_json_sql_path_is_valid (std::string &sql_path, bool allow_wildcards)
 	      break;
 	    default:
 	      // unquoted object_keys
+	      // we normalize object_keys by quoting them - e.g. $."objectkey" will be representant for $.objectkey
+	      sql_path.insert (i++, "\"");
```

Comment:
Move `sql_path.insert (i++, "\"");` to db_json_path_is_token_valid_unquoted_object_key.